graph/api/src/main/java/org/jboss/windup/graph/model/resource/FileModel.java

```diff
@@ -322,5 +327,11 @@ public File asFile() throws RuntimeException
 
             return new File(getFilePath());
         }
+
+        @Override
+        public boolean belongsToProject(ProjectModel projectModel)
+        {
+            return this.getProjectModel().equals(projectModel);
```

Comment:
You might want to verify that the project being passed in is a canonical project first. Check if it is an instance of DuplicateProjectModel and if it is, then use the result of calling getCanonicalProject() on it.